### ISSUE TITLE
archlinuxarm armv7h problem compiling 

### ISSUE THREAD

**Issue by enricoT** [way-cooler/way-cooler#445]
Im trying to build on archlinuxarm for raspberry pi 3b armv7h in arch PKGBUILD  file i added this line:

build() {
  cd "$srcdir/$pkgname"
....  line
  cargo check -vv
.... end line
  cargo build --release
}

```
     Running `rustc --crate-name rustwlc /home/alarm/.cargo/registry/src/github.com-1ecc6299db9ec823/rustwlc-0.6.4/src/lib.rs --crate-type lib --emit=dep-info,metadata -C debuginfo=2 --cfg 'feature="wayland-sys"' --cfg 'feature="wlc-wayland"' -C metadata=9e88d99c0ef509c2 -C extra-filename=-9e88d99c0ef509c2 --out-dir /tmp/yaourt-tmp-alarm/aur-way-cooler-git/src/way-cooler-git/target/debug/deps -L dependency=/tmp/yaourt-tmp-alarm/aur-way-cooler-git/src/way-cooler-git/target/debug/deps --extern libc=/tmp/yaourt-tmp-alarm/aur-way-cooler-git/src/way-cooler-git/target/debug/deps/liblibc-6e25f4998d90c5b0.rmeta --extern bitflags=/tmp/yaourt-tmp-alarm/aur-way-cooler-git/src/way-cooler-git/target/debug/deps/libbitflags-31bc10994438d7fa.rmeta --extern wayland_sys=/tmp/yaourt-tmp-alarm/aur-way-cooler-git/src/way-cooler-git/target/debug/deps/libwayland_sys-777c3956644e2f91.rmeta --cap-lints warn`
     Running `rustc --crate-name wayland_server /home/alarm/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-server-0.9.10/src/lib.rs --crate-type lib --emit=dep-info,metadata -C debuginfo=2 -C metadata=c091bf60917268e2 -C extra-filename=-c091bf60917268e2 --out-dir /tmp/yaourt-tmp-alarm/aur-way-cooler-git/src/way-cooler-git/target/debug/deps -L dependency=/tmp/yaourt-tmp-alarm/aur-way-cooler-git/src/way-cooler-git/target/debug/deps --extern bitflags=/tmp/yaourt-tmp-alarm/aur-way-cooler-git/src/way-cooler-git/target/debug/deps/libbitflags-3b38584a7f6a3081.rmeta --extern nix=/tmp/yaourt-tmp-alarm/aur-way-cooler-git/src/way-cooler-git/target/debug/deps/libnix-e8f92881ecc998ca.rmeta --extern libc=/tmp/yaourt-tmp-alarm/aur-way-cooler-git/src/way-cooler-git/target/debug/deps/liblibc-6e25f4998d90c5b0.rmeta --extern wayland_sys=/tmp/yaourt-tmp-alarm/aur-way-cooler-git/src/way-cooler-git/target/debug/deps/libwayland_sys-4597307585999c55.rmeta --cap-lints warn`
warning: unused `#[macro_use]` import
  --> /home/alarm/.cargo/registry/src/github.com-1ecc6299db9ec823/rustwlc-0.6.4/src/lib.rs:67:1
   |
67 | #[macro_use]
   | ^^^^^^^^^^^^
   |
   = note: #[warn(unused_imports)] on by default

error[E0308]: mismatched types
   --> /home/alarm/.cargo/registry/src/github.com-1ecc6299db9ec823/rustwlc-0.6.4/src/handle.rs:706:21
    |
706 |             chars = wlc_view_get_title(self.0);
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^ expected i8, found u8
    |
    = note: expected type `*const i8`
               found type `*const u8`
    = help: here are some functions which might fulfill your needs:
            - .offset(...)
            - .wrapping_offset(...)

error[E0308]: mismatched types
   --> /home/alarm/.cargo/registry/src/github.com-1ecc6299db9ec823/rustwlc-0.6.4/src/handle.rs:710:39
    |
710 |                     pointer_to_string(chars)
    |                                       ^^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`
    = help: here are some functions which might fulfill your needs:
            - .offset(...)
            - .wrapping_offset(...)

error[E0308]: mismatched types
   --> /home/alarm/.cargo/registry/src/github.com-1ecc6299db9ec823/rustwlc-0.6.4/src/handle.rs:719:21
    |
719 |             chars = wlc_view_get_class(self.0);
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^ expected i8, found u8
    |
    = note: expected type `*const i8`
               found type `*const u8`
    = help: here are some functions which might fulfill your needs:
            - .offset(...)
            - .wrapping_offset(...)

error[E0308]: mismatched types
   --> /home/alarm/.cargo/registry/src/github.com-1ecc6299db9ec823/rustwlc-0.6.4/src/handle.rs:723:35
    |
723 |                 pointer_to_string(chars)
    |                                   ^^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`
    = help: here are some functions which might fulfill your needs:
            - .offset(...)
            - .wrapping_offset(...)

error[E0308]: mismatched types
   --> /home/alarm/.cargo/registry/src/github.com-1ecc6299db9ec823/rustwlc-0.6.4/src/handle.rs:732:21
    |
732 |             chars = wlc_view_get_app_id(self.0);
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected i8, found u8
    |
    = note: expected type `*const i8`
               found type `*const u8`
    = help: here are some functions which might fulfill your needs:
            - .offset(...)
            - .wrapping_offset(...)

error[E0308]: mismatched types
   --> /home/alarm/.cargo/registry/src/github.com-1ecc6299db9ec823/rustwlc-0.6.4/src/handle.rs:736:35
    |
736 |                 pointer_to_string(chars)
    |                                   ^^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`
    = help: here are some functions which might fulfill your needs:
            - .offset(...)
            - .wrapping_offset(...)

error: aborting due to 6 previous errors

error: Could not compile `rustwlc`.

Caused by:
  process didn't exit successfully: `rustc --crate-name rustwlc /home/alarm/.cargo/registry/src/github.com-1ecc6299db9ec823/rustwlc-0.6.4/src/lib.rs --crate-type lib --emit=dep-info,metadata -C debuginfo=2 --cfg feature="wayland-sys" --cfg feature="wlc-wayland" -C metadata=9e88d99c0ef509c2 -C extra-filename=-9e88d99c0ef509c2 --out-dir /tmp/yaourt-tmp-alarm/aur-way-cooler-git/src/way-cooler-git/target/debug/deps -L dependency=/tmp/yaourt-tmp-alarm/aur-way-cooler-git/src/way-cooler-git/target/debug/deps --extern libc=/tmp/yaourt-tmp-alarm/aur-way-cooler-git/src/way-cooler-git/target/debug/deps/liblibc-6e25f4998d90c5b0.rmeta --extern bitflags=/tmp/yaourt-tmp-alarm/aur-way-cooler-git/src/way-cooler-git/target/debug/deps/libbitflags-31bc10994438d7fa.rmeta --extern wayland_sys=/tmp/yaourt-tmp-alarm/aur-way-cooler-git/src/way-cooler-git/target/debug/deps/libwayland_sys-777c3956644e2f91.rmeta --cap-lints warn` (exit code: 101)
warning: build failed, waiting for other jobs to finish...
warning: missing documentation for a struct
   --> /home/alarm/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-server-0.9.10/src/event_loop.rs:334:1
    |
334 | / pub struct EventLoop {
335 | |     ptr: *mut wl_event_loop,
336 | |     display: Option<*mut wl_display>,
337 | |     handle: Box<EventLoopHandle>,
338 | | }
    | |_^
    |
note: lint level defined here
   --> /home/alarm/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-server-0.9.10/src/lib.rs:176:9
    |
176 | #![warn(missing_docs)]
    |         ^^^^^^^^^^^^

error: build failed
==> ERROR: A failure occurred in build().
    Aborting...
==> ERROR: Makepkg was unable to build way-cooler-git.
==> Restart building way-cooler-git ? [y/N]

```

**Comment by Timidger** [comment]
Hi there, thanks for the report!

These issues exist rust-wlc. These issues are also preventing #369, so I'll probably fix this in the coming day. Thanks for the detailed error message, as I don't have a ARM chip to test my code on (and setting up cross-compilation has never ended well for me or my system...)

**Comment by Timidger** [comment]
Well it turns out [cross](https://github.com/japaric/cross) is bloody amazing, so I'm very sure that I have fixed this issue. Will update rust-wlc, ensure it works in Way Cooler, and then bump its version.

**Comment by Timidger** [comment]
Ok it works. I'll update Way Cooler once I regain crates.io access (somehow accidentally messed up my access...)

**Comment by Timidger** [comment]
This will be fixed when #453 is merged in (it includes a dependency bump that will fix this)

**Comment by Timidger** [comment]
Fixed
